river/command/map.zig

```diff
@@ -251,6 +258,10 @@ pub fn unmap(
     const mode_mappings = &server.config.modes.items[mode_id].mappings;
     const mapping_idx = mappingExists(mode_mappings, modifiers, keysym, optionals.release) orelse return;
 
+    // Repeating mappings borrow the args from Mapping directly. To prevent a crash in the
+    // unlikely case of a repeating mapping unmapping itself, clear any current repeat.
+    seat.clearRepeatingMapping();
```

Comment:
Yes, I cheated there. Doing an unmap while repeating a key mapping seemed like a unlikely edge case, but I certainly didn't want it to cause a crash.

Reply:
Wouldn't an unmap cause elements to shift in the ArrayList so a stored index would potentially point to the wrong Mapping (or even off the end of the ArrayList)? I could just save a copy of the repeating mapping but I was (perhaps unhelpfully?) trying to avoid copying.

Reply:
> Wouldn't an unmap cause elements to shift in the ArrayList so a stored index would potentially point to the wrong Mapping (or even off the end of the ArrayList)? I could just save a copy of the repeating mapping but I was (perhaps unhelpfully?) trying to avoid copying.

Oh you're totally right, how could I forget about unmapping. I think the way you have it is good then, I agree that we should avoid the copy.